### DISCUSSION TITLE
Median vs mean in pylinac CT QA

### DISCUSSION THREAD

**Discussion by spaynemonash** [jrkerns/pylinac#472] [General]
Hi everyone,

Me and my team have started using pylinac for implementing CT QC at our hospital, however we have run into a problem using the pylinac code.

Namely ROI values for HU are based on median values, however our regional CT compliance legislation is specifically in terms of mean values. This can cause us to erroneously fail criteria using pylinac due to the differences between median and mean values in ROI.

Is this something that could be implemented into the program? Ideally one would be able to change between mean and median, which would allow for comparisons between the two types of averages. And that way those who prefer to work with medians could do so.

**Comment by jrkerns** [comment]
Right now you can get the mean value of an ROI like so:

```
import numpy as np
from pylinac import CatPhan504

ct = CatPhan504.from_demo_images()
ct.analyze()
air_mean = np.mean(ct.ctp404.rois['Air'].pixel_values)
```

This could also be added as a property to the ROIs in the library (e.g. `ct.ctp404.rois['Air'].mean`), but depends on how you're using the library. I'm guessing you mean something different?

**Comment by nitschkek** [comment]
Hi James,
I have just finished analysing HU constancy historical data for the Gammex 467 phantom with Pylinac using a custom cheese phantom class (that removes the couch from the CT data so the phantom center finding code works). I have compared the results with Doselab which uses the ROI mean HU. It would be nice if the phantom class/module could specify which metric (mean or median) to use for the analysis ... where ever the DiskROI class is used (i.e. Catphan or cheese phantoms). I would be keen to code such a modification. Is this something you would consider incorporating in the package? Otherwise I will just add it to my custom class for the Gammex 467 in the manner you have suggested.
Thanks
Kym

**Reply by jrkerns** [reply]
Although it would be nice to have, it would be considerably more work to add it to all CT-like analysis routines and our backlog is pretty long right now. I'd add it to your custom class and see how it works out first.

**Reply by nitschkek** [reply]
Thanks James. I'll continue working on adding the ROI mean HU to my custom classes. I'll share the code once I have used it with the Gammex 467 and Catphan CTP 401/4 module historical analysis to see what you think.
